pkg/config/mock.go

```diff
@@ -0,0 +1,27 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package config
+
+import "strings"
+
+// MockConfig should only be used in tests
+type MockConfig struct {
+	Config
+}
+
+// Set is used for setting configuration in tests
+func (c *MockConfig) Set(key string, value interface{}) {
+	c.Config.Set(key, value)
+}
+
+// NewMock is creating and returning a mock config
+func NewMock() *MockConfig {
```

Comment:
This confused me initially because it gives the impression that it simply returns a mock config and it's not immediately obvious that it replaces the global one, so I'd suggest:
```go
// Mock replaces the global config with a mock one and returns an interface to allow
// editing it.
func Mock() (*MockConfig)
```
I'm also wondering if we should have a second return parameter which would restore any configuration that was replaced by this call.

Reply:
+1 on changing the name of the function to `Mock` (unless there's a good reason not to), and mentioning that it replaces the global config

Reply:
I agree it's clearer 👍